examples/mysql-kopia-repository-server/README.md

```diff
@@ -0,0 +1,429 @@
+
+You can retrieve your root password by running the following command. Make sure to replace [YOUR_RELEASE_NAME] and [YOUR_NAMESPACE]:
+
+    `kubectl get secret [YOUR_RELEASE_NAME] --namespace [YOUR_NAMESPACE] -o jsonpath="{.data.mysql-root-password}" | base64 --decode`
+
+> **Tip**: List all releases using `helm list --all-namespaces`, using Helm Version 3.
+
+## Integrating with Kanister
+
+If you have deployed MySQL application with name other than `mysql-release` and namespace other than `mysql`, you need to modify the commands(backup, restore and delete) used below to use the correct release name and namespace
+
+### Create Repository Server CR
+
+Create Kopia Repository Server CR if not created already
+
+- Create Kopia Repository
+```bash
+$ kopia --log-level=error --config-file=/tmp/kopia-repository.config \
+--log-dir=/tmp/kopia-cache repository create --no-check-for-updates \
+--cache-directory=/tmp/cache.dir --content-cache-size-mb=0 \
+--metadata-cache-size-mb=500 --override-hostname=mysql.app \
+--override-username=kanisterAdmin s3 --bucket=<s3_bucket_name> \
+--prefix=/repo-controller/ --region=<s3_bucket_region> \
+--access-key=<aws_access_key> --secret-access-key=<aws_secret_access_key> --password=<repository_password>
+```
+
+- Generate TLS Certificates and create TLS secret for Kopia Repository Server for secure communication between Kopia Repository Server and Client
+
+```bash
+$ openssl req -newkey rsa:2048 -nodes -keyout key.pem -x509 -days 365 -out certificate.pem
+
+$ kubectl create secret tls repository-server-tls-cert --cert=certificate.pem --key=key.pem -n kanister
+```
+
+- Create Location Secrets for Kopia Repository
+
+```bash
+# The following file s3_location_creds.yaml is a sample file for creating s3 credentials secrets. It contains the credentials for accessing the s3 bucket.
+$ vi s3_location_creds.yaml
+
+apiVersion: v1
+kind: Secret
+metadata:
+   name: s3-creds
+   namespace: kanister
+   labels:
+      repo.kanister.io/target-namespace: monitoring
+type: secrets.kanister.io/aws
+data:
+   # required: base64 encoded value for key with proper permissions for the bucket
+   aws_access_key_id: <base64_encoded_aws_access_key>
+   # required: base64 encoded value for the secret corresponding to the key above
+   aws_secret_access_key: <base64_encoded_aws_secret_access_key>
+```
+
+```
+$ kubectl create -f s3_location_creds.yaml -n kanister
+```
+
+```bash
+# The following file s3_location.yaml is a sample file for creating s3 location secrets. It contains the details of the s3 bucket.
+$ vi s3_location.yaml
+
+apiVersion: v1
+kind: Secret
+metadata:
+   name: s3-location
+   namespace: kanister
+   labels:
+      repo.kanister.io/target-namespace: monitoring
+type: Opaque
+data:
+   # required: specify the type of the store
+   # supported values are s3, gcs, azure, and file-store
+   type: czM=
```

Comment:
why are we specifying that this is base 64 encoded as we are specifying for other keys in this secret?

Reply:
its base64 encoded value of s3

Reply:
My bad, we are using s3, hence the base64 encoded value. can you add a comment over there specifying its base64 encoded value of s3